### ISSUE TITLE
Amplify Build is broken with error - "TypeScript executable not found."

### ISSUE THREAD

**Issue by KarthickSelvam** [aws-amplify/amplify-hosting#3231]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d243offqyjitf0

### AWS Region

eu-central-1

### Amplify Hosting feature

Backend builds

### Describe the bug

the deployment CI is failing with error - 

2023-01-09T10:37:30.981Z [INFO]: [0mAmplify AppID found: d243offqyjitf0. Amplify App name is: ----[0m
2023-01-09T10:37:31.032Z [INFO]: [0mBackend environment test found in Amplify Console app: -----[0m
2023-01-09T10:37:31.899Z [WARNING]: - Fetching updates to backend environment: test from the cloud.
2023-01-09T10:37:32.760Z [WARNING]: - Building resource auth/-------
2023-01-09T10:37:32.761Z [WARNING]: - Building resource api/-----
2023-01-09T10:37:32.762Z [WARNING]: - Building resource custom/----
2023-01-09T10:37:32.762Z [WARNING]: - Building custom resources
2023-01-09T10:37:38.093Z [WARNING]: ✖ There was an error initializing your environment.
2023-01-09T10:37:38.563Z [INFO]: 🛑 TypeScript executable not found.
                                 Resolution: Please add it as a dev-dependency in the package.json file for this resource.
2023-01-09T10:37:38.564Z [INFO]: Learn more at: https://docs.amplify.aws/cli/project/troubleshooting/
2023-01-09T10:37:38.565Z [INFO]: 
2023-01-09T10:37:38.565Z [INFO]: Session Identifier: 14ff5981-c8aa-4926-bf52-c625f18f47c3

### Expected behavior

Build completes without error, even previous successful build is failing now so that should pass.

### Reproduction steps

Go to the console
Go to the amplify app
click on "redeploy this version" and see the build is failing

### Build Settings

```yaml
version: 1
backend:
  phases:
    build:
      commands:
        - npm i -g graphql-ttl-transformer
        - amplifyPush --simple
frontend:
  phases:
    preBuild:
      commands:
        - yarn install
    build:
      commands:
        - yarn run build
  artifacts:
    baseDirectory: build
    files:
      - "**/*"
  cache:
    paths:
      - node_modules/**/*
test:
  artifacts:
    baseDirectory: cypress
    configFilePath: "**/mochawesome.json"
    files:
      - "**/*.png"
      - "**/*.mp4"
  phases:
    preTest:
      commands:
        - yarn install
        - yarn add mocha mochawesome mochawesome-merge mochawesome-report-generator
    test:
      commands:
        - npx start-test 'yarn start' 3000 'npx cypress run --reporter mochawesome --reporter-options "reportDir=cypress/report/mochawesome-report,overwrite=false,html=false,json=true,timestamp=mmddyyyy_HHMMss"'
    postTest:
      commands:
        - npx mochawesome-merge cypress/report/mochawesome-report/mochawesome*.json > cypress/report/mochawesome.json
```

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by anurag128** [comment]
Can you try pinning down to amplify CLI version 10.6.0 in [live package updates](https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-image.html#setup-live-updates) from build settings.

**Comment by jerushachua** [comment]
I'm using amplify cli version `10.6.1`

**Comment by josefaidt** [comment]
Hey @jerushachua :wave: as we look at this issue in more depth, can you confirm TypeScript is noted as a devDependency for your custom CDK resource? The CLI appears to be erroring looking for the `tsc` executable at `amplify/backend/custom/<resource-name>/node_modules/.bin/tsc`

**Comment by jerushachua** [comment]
I have four custom resources configured for this project, and checked the package.json for all them - they all specify typescript as a dev dependency: 

```
"devDependencies": {
    "typescript": "^4.2.4"
  }
```

**Comment by KarthickSelvam** [comment]
Yes same here, 

  "devDependencies": {
    "typescript": "^4.2.4"
  }
  
  Also please note this is not the first build, the resources are already created, and the build was working fine so far.

**Comment by KarthickSelvam** [comment]
> Can you try pinning down to amplify CLI version 10.6.0(or to a version you are using locally) in [live package updates](https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-image.html#setup-live-updates) from build settings.

yes tried with the 10.6.0 faced same error. 

Btw my local cli is running 10.6.1 and I'm able to do Amplify push. The issue is only on the CI build.

**Comment by KarthickSelvam** [comment]
> Can you try pinning down to amplify CLI version 10.6.0(or to a version you are using locally) in [live package updates](https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-image.html#setup-live-updates) from build settings.

yes tried with the 10.6.0 faced same error. 

Btw my local cli is running 10.6.1 and I'm able to do Amplify push from local, the issue is only on the CI build. And its currently blocking a production release! Looking forward for a resolution soon..

**Comment by KarthickSelvam** [comment]
Hi team, do we have any workaround on this?

**Comment by KarthickSelvam** [comment]
Adding this to the build file worked

backend:
  phases:
    preBuild:
        commands:
          - yarn install

**Comment by KarthickSelvam** [comment]
But the second build after that failed, but changing the ts version in the package file worked again.. seems very inconsistent behaviour
![image](https://user-images.githubusercontent.com/889316/214021223-77c5a0d5-ecc4-47e3-9823-323ba7d22410.png)

**Comment by KarthickSelvam** [comment]
Summary of the issue and debugging so far: 

- Adopted CDK for custom resources and did a successful deployment
- Following deployment failed
- tried adding ts in dev dependency in the root project
- added yarn install to the amplify.yml (never needed this before) in the backend prebuild step - deployment success
- following deployment failed
- modified ts version from latest to 4.2.4 in the root project- deployment success
- following deployment failed
- modifying ts version to latest in the root project - deployment success
-  it is weird but it seems like every time it needs some change to the ts version in the package.json

Hope this helps to identify the issue!

**Comment by josefaidt** [comment]
Hey @KarthickSelvam and folks in the thread :wave: this issue should now be resolved with the release of Amplify CLI 10.7.0, can you try upgrading and rebuilding to see if this resolves your issue?

**Comment by josefaidt** [comment]
Hey @KarthickSelvam just wanted to follow up on this one and see if you're still experiencing this issue when using the latest version of Amplify CLI (10.7.1)?

**Comment by hloriana** [comment]
This issue should be resolved with Amplify CLI 10.7.0, please open a new issue if you continue to run into this behavior. Thank you!
